scripts/get_oppia_classes.py

```diff
@@ -0,0 +1,196 @@
+The script will search among all the files that lie below that directory
+in file structure i.e it will search in subdirs recursively also.
+
+====================================
+CUSTOMIZATION OPTIONS FOR DEVELOPERS
+====================================
+You can also use this script for similar use cases but different type of
+files by changing the variables html_pattern, directive_pattern and class
+_pattern respectively
+"""
+
+import os
+import re
+import bs4
+
+
+def return_all_files(dir_to_scan):
+    """
+    returns all files that lie below that directory in file structure
+
+    ARGS: directory name
+    RETURNS: all files in the path
+    """
+    file_result = []
+    subdirs = []
+    for root, subdir, files in os.walk(dir_to_scan):
+        for f in files:
+            file_result.append(os.path.join(root, f))
+        for d in subdir:
+            subdirs.append(d)
+    return file_result
+
+
+def return_matched_files(files, pattern):
+    """
+    returns all the files matching given pattern
+
+    ARGS: list of files to search from, pattern to search
+    RETURNS: list of files matching pattern
+    """
+    val = []
+    for f in files:
+        if pattern.match(str(f)) is not None:
+            val.append(f)
+    return val
+
+
+def match_classes_pattern(classes, pattern):
+    """
+    return the classes which match given pattern
+
+    ARGS: list of classes to search from, pattern
+    RETURNS: list of classes matching pattern
+    """
+    matched_classes = []
+    if classes is None:
+        return matched_classes
+    else:
+        for j in classes:
+            if pattern.match(j):
+                matched_classes.append(j)
+    return matched_classes
+
+
+def file_find_matching_classes(file_name, pattern):
+    """
+    returns all the classes matching the pattern in a file
+
+    ARGS: file name, pattern to match
+    RETURNS: list of classes matching pattern
+    """
+    f = open(file_name)
+    # making BeautifulSoup object of file then extracting all classes
+    # and if the tag is script we further make BeautifulSoup object to
+    # take case of ng-template
+    file_soup = bs4.BeautifulSoup(f.read(), 'html.parser')
+
+    list_oppia_classes = []
+
+    for i in file_soup.findAll():
+        if i.name == 'script' and i.attrs.get('type') == 'text/ng-template':
+            inside_script_content = str(i.contents)
+            soup_from_text_inside_script = bs4.BeautifulSoup(
+                inside_script_content, 'html.parser')
+            for t in soup_from_text_inside_script.findAll():
+                cur_classes = t.attrs.get('class')
+                list_oppia_classes.extend(
+                    match_classes_pattern(cur_classes, pattern))
+        else:
+            cur_classes = i.attrs.get('class')
+            list_oppia_classes.extend(
+                match_classes_pattern(cur_classes, pattern))
+
+    return set(list_oppia_classes)
+
+
+def print_output(file_dict):
```

Comment:
print_dict or pretty_print_dict

(this function should not need to know what the dict represents, hence no need for "output")